### ISSUE TITLE
early_stopping does not work as expected 

### ISSUE THREAD

**Issue by 5tinzi** [FluxML/Flux.jl#2089]
### Package Version

 Flux v0.13.6

### Julia Version

julia v1.7

### OS / Environment

macOS 12.1 21C52 x86_64

### Describe the bug

Flux has a function for early stopping, as describes in the documents page here: [Utility Functions · Flux](https://fluxml.ai/Flux.jl/stable/utilities/#Flux.early_stopping)

    For example, you can use early_stopping to stop training when the model is converging or deteriorating

    By default distance is `-`, which implies that the monitored metric f is expected to be decreasing and minimized. 

But actually, the `early_stop` function will  stop the train even if the loss is keeping decreasing!

### Steps to Reproduce

```julia
julia> loss = let l = 10
         () -> l -= 1
       end; # NOTICE this loss function that returns decreasing values
julia> es = Flux.early_stopping(loss, 3);
julia> for i in 1:10
         @info "Epoch $i"
         es() && break
       end
[ Info: Epoch 1
[ Info: Epoch 2
[ Info: Epoch 3
```

### Expected Results

the training won't be stop if the loss is keeping decreasing.

### Observed Results

the training step was stop even the loss is keeping decreasing.
